### ISSUE TITLE
expose yaml schema store

### ISSUE THREAD

**Issue by jmbuhr** [quarto-dev/quarto#195]
Over in the [quarto neovim integration](https://github.com/quarto-dev/quarto-nvim) I am now close to introducing a feature that allows for code completion etc. in all the embedded languages i.e. not just code chunks but also raw `html` and `yaml`: https://github.com/jmbuhr/otter.nvim/pull/39 (It's very exciting!)

In order for the yaml completion to be useful, I would need a schema of possible yaml keys. I found https://github.com/quarto-dev/quarto-cli/tree/main/src/resources/schema, but this is split across many files. Ideally it would be something along the lines of what https://www.schemastore.org/json/, which are often used for the yaml language server https://github.com/redhat-developer/yaml-language-server#using-yamlschemas-settings

**Comment by cscheid** [comment]
We unfortunately can't; our schema is genuinely too complicated and it's derived dynamically at the start of the IDE (we also have features that are essential for our error reporting that are not compatible with JSON schema).

**Comment by cscheid** [comment]
With that said, we should eventually have a way for you to request "complete YAML from this point" through the quarto LSP. That, I believe, will do precisely what you need. So stay tuned!

**Comment by jjallaire** [comment]
Yes, we are working on having a Quarto LSP that you should be able to pickup all of the YAML completions from (as well as hover support for math and completions for various other div classes and attributes). We want to make this available via `quarto lsp` however this isn't currently technically possible (but should be in future versions of deno as they get better node compatibility).
